docs/proposals/scheduling/multi-podtemplate-support/multiple-pod-template-support.md

```diff
@@ -0,0 +1,195 @@
+    TotalReplicas int32 `json:"totalReplicas,omitempty"`
+
+    // Defines the requirements of an individual component of the resource.
+	// +optional
+    Components []Components `json:"components,omitempty"`
+
+	. . .
+}
+
+// A component is a unique representation of a resource's replica. For simple resources, like Deployments, there will only be
+// one component, associated with the podTemplate in the Deployment definition.
+//
+// Complex resources can have multiple components controlled through different podTemplates.
+// Each replica for the resource will fall into a component type with requirements defined by its relevant podTemplate.
+type ComponentRequirements struct {
+
+    // Name of this component
+    Name string `json:"name,omitempty"`
+
+    // Replicas represents the replica number of the resource's component
+    // +optional
+	Replicas int32 `json:"replicas,omitempty"`
+
+    // ReplicaRequirements represents the requirements required by each replica for this component.
+	// +optional
+	ReplicaRequirements *ReplicaRequirements `json:"replicaRequirements,omitempty"`
+
+}
+
+// ReplicaRequirements represents the requirements required by each replica.
+type ReplicaRequirements struct {
+
+	// NodeClaim represents the node claim HardNodeAffinity, NodeSelector and Tolerations required by each replica.
+	// +optional
+	NodeClaim *NodeClaim `json:"nodeClaim,omitempty"`
+
+	// ResourceRequest represents the resources required by each replica.
+	// +optional
+	ResourceRequest corev1.ResourceList `json:"resourceRequest,omitempty"`
+
+	// Namespace represents the resources namespaces
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// PriorityClassName represents the components priorityClassName
+	// +optional
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+
+}
+```
+
+### Overview of Code Changes
+
+Besides the change to the `ResourceBindingSpec` and the `ReplicaRequirements API`, we will need to make a code change to the accurate estimator's implementation,
+which can be found here: https://github.com/karmada-io/karmada/blob/5e354971c78952e4f992cc5e21ad3eddd8d6716e/pkg/estimator/server/estimate.go#L59, as well as the maxReplica estimation done by the ResourceQuota plugin.
+
+Currently the accurate estimator will calculate the maxReplica count by:
+1. Running the maxReplica calculation (`maxReplicas`) for each plugin enabled by the accurate estimator.
+2. The accurate estimator will then loop through all nodes and sum up the amount of replicas (`sumReplicas`) that can fit in each node. This is to account for the resource fragmentation issue.
+3. The result returned will be: `Math.min(maxReplicas, sumReplicas)`.
+
+![Accurate-Scheduler-Steps](Accurate-Scheduler-Steps.png)
+
+For the proposed implementation, please refer to the next section.
+
+### Accurate Estimator Changes
```

Comment:
Apologies if the explanation of the implementation proposal is a little bit dense. I can format this to be a little more clear, but otherwise please let me know if you have any questions that I can clarify.